master/pkg/searcher/asha.go

```diff
@@ -170,7 +171,30 @@ func (s *asyncHalvingSearch) trialClosed(
 ) ([]Operation, error) {
 	s.TrialsCompleted++
 	s.ClosedTrials[requestID] = true
-	return nil, nil
+	var ops []Operation
+	if s.sendShutdownOperation() {
```

Comment:
Please note we don't need do this in trialExitedEarly in ASHA because if the last trial did not exit because of a HP related reason then it will be marked as a EarlyExitedTrial and it will call s.promoteAsync where this trial will be closed. Please see: https://github.com/determined-ai/determined/blob/d42ada95bc3d6a334f3322b3c9c6755085844b8a/master/pkg/searcher/asha.go#L345.